content/docs/guides/neon-authorize.md

```diff
@@ -0,0 +1,150 @@
+---
+title: About Neon Authorize
+subtitle: Secure your application at the database level using Postgres's Row-Level Security
+enableTableOfContents: true
+---
+
+<InfoBlock>
+<DocsList title="What you will learn:">
+<p>JSON Web Tokens (JWT)</p>
+<p>Row-level Security (RLS)</p>
+<p>How Neon Authorize works</p>
+</DocsList>
+
+<DocsList title="Before you start" theme="docs">
+<a href="/docs/guides/neon-authorize-quickstart">Get started</a>
+</DocsList>
+
+<DocsList title="Example repository" theme="repo">
+<a href="https://github.com/orgs/neondatabase/repositories?type=source&q=authorize">6 sample demos</a>
+</DocsList>
+</InfoBlock>
+
+<ComingSoon/>
+
+**Neon Authorize** integrates with third-party **JWT-based authentication providers** like Auth0 and Clerk, bringing authorization closer to your data by leveraging **Row-Level Security (RLS)** at the database level.
+
+## Authentication and authorization
+
+When implementing user authentication in your application, third-party authentication providers like **Clerk**, **Auth0**, and others simplify the process of managing user identities, passwords, and security tokens. Once a user's identity is confirmed, the next step is **authorization** — controlling who can do what in your app based on their user type or role — for example, admins versus regular users. With Neon Authorize, you can handle authorization right in Postgres, in addition to, or to replace entirely, security at other layers.
+
+## How Neon Authorize works
+
+Most authentication providers issue **JSON Web Tokens (JWTs)** on user authentication to convey user identity and claims. The JWT is a secure way of proving that logged-in users are who they say they are &#8212; and passing that proof on to other entities.
+
+With **Neon Authorize**, the JWT is passed on to Neon, where you can make use of the validated user identity right in Postgres. To integrate with an authentication provider, add your provider's JWT discovery URL to your project. This lets Neon retrieve the necessary keys to validate the JWTs.
+
+Behind the scenes, the [Neon Proxy](#the-role-of-the-neon-proxy) performs the validation, while the open source extension [pg_session_jwt](#how-the-pg_session_jwt-extension-works) makes the extracted `user_id` available to Postgres. You can then use **Row-Level Security (RLS)** policies in Postgres to enforce access control at the row level, ensuring that users can only access or modify data according to the defined rules. Since these rules are implemented directly in the database, they can offer a secure fallback — or even a primary solution — in case security in other layers of your application fail. See [when to rely on RLS](#when-to-rely-on-rls) for more information.
+
+![neon authorize architecture](/docs/guides/neon_authorize_architecture.png)
+
```

Comment:
Should we add a note to mention that it's possible to use Neon Auth without an auth provider, as mentioned here?
https://neondb.slack.com/archives/C066CAC2NQK/p1729203665691569

Reply:
added